### PR TITLE
RBAC mechanism

### PR DESCRIPTION
**Pull request by lebauce**

### CODE REVIEW

**Comment by lebauce** [comment]
run  skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run skydive-functional-tests-backend-elasticsearch

**Comment by lebauce** [comment]
run  skydive-selenium-tests

**Comment by lebauce** [comment]
run skydive-selenium-tests

**Comment by lebauce** [comment]
run tests

File: rbac/etcdadapter.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+	"github.com/casbin/casbin/model"
+	"github.com/casbin/casbin/persist"
+	"github.com/casbin/casbin/util"
+	etcd "github.com/coreos/etcd/client"
+)
+
+const EtcdPolicyKey = "/casbinPolicy"
+
+// EtcdAdapter represents the etcd adapter for policy persistence, can load policy from consul or save policy to consul.
```
[the remark this answers is not in the document]
Comment (lebauce) [reply rbac/etcdadapter.go]: Fixed

File: statics/js/components/graph-layout.js [hunk 2]
```diff
@@ -1332,23 +1330,16 @@ TopologyGraphLayout.prototype = {
...
 
-  getLocalValue: function(key) {
-    if (!localStorage.preferences) return 0;
-    var v = localStorage.preferences[key];
-    if (!v || v === "0" || v === "null") return 0;
-    return Number(v);
-  },
-
   loadBandwidthConfig: function() {
     var vm = this.vm, b = this.bandwidth, self = this;
```
[the remark this answers is not in the document]
Comment (lebauce) [reply statics/js/components/graph-layout.js]: Removed

File: statics/js/app.js [hunk 3]
```diff
@@ -282,16 +280,57 @@ var app = new Vue({
...
+        }
+        return value;
+      }
+    },
+
+    enforce: function(subject, action) {
+      var self = this;
+
+      var perms = self.$store.state.permissions;
+      for (perm in perms) {
```
[the remark this answers is not in the document]
Comment (lebauce) [reply statics/js/app.js]: Fixed

File: statics/js/app.js [hunk 4]
```diff
@@ -282,16 +280,57 @@ var app = new Vue({
...
+        if (perms[perm][1] === subject &&
+            perms[perm][2] == action) {
```
[the remark this answers is not in the document]
Comment (lebauce) [reply statics/js/app.js]: Fixed

**Comment by lebauce** [comment]
@eranra You can set the `rbac.policy` configuration parameter (or the SKYDIVE_RBAC_POLICY environment variable) with "p, admin, capture, write, deny" to completely disable a feature when using no authentication backend

**Comment by eranra** [comment]
@lebauce, I was looking on skydive.yml.default to try and understand  the details of "p, admin, capture, write, deny" and it is a bit of a puzzle without more explanations. Can yo add more remarks/documentation  so it will be easier to follow.

**Comment by lebauce** [comment]
@eranra There is a pending pull request on the Skydive documentation regarding RBAC : https://github.com/skydive-project/skydive.network/pull/11/

**Comment by eranra** [comment]
Good ... looks much better maybe reference from the default etc file ... also 'p', and 'g' you might want to explain what they do  and what is the complete list of options for 'object's

**Comment by lebauce** [comment]
@eranra The documentation links to the bundle policy that "contains all the possible objects" but, as the PR is not merged yet, the link does not exist yet. You can use this one for now : https://github.com/lebauce/skydive/blob/rbac/rbac/policy.csv

**Comment by eranra** [comment]
Perfect 👍

**Pull request merged**